domain/consensus/model/pow/pow.go

```diff
@@ -0,0 +1,55 @@
+package pow
+
+import (
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/hashes"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/serialization"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/pkg/errors"
+	"math/big"
+)
+
+// CheckProofOfWork check's if the block has a valid PoW according to its Bits field
+// it does not check if the difficulty itself is valid or less than the maximum for the appropriate network
+func CheckProofOfWork(header *externalapi.DomainBlockHeader, optionalTarget *big.Int) bool {
+	if optionalTarget == nil {
+		optionalTarget = util.CompactToBig(header.Bits)
+	}
+	// The block hash must be less than the claimed target unless the flag
+	// to avoid proof of work checks is set.
+	powNum := hashes.ToBig(calcPowValue(header))
+
+	// The block hash must be less or equal than the claimed target.
+	return powNum.Cmp(optionalTarget) <= 0
+}
+
+func calcPowValue(header *externalapi.DomainBlockHeader) *externalapi.DomainHash {
```

Comment:
Correct me if I misunderstood the code here... seems to me this means that the block hash returned by consensushashing (and therefore used in all APIs, etc.), and the block hash that is used for validating PoW are different.

Do we really want this?

I'd rather if hashserialization.BlockHash did all this as well.

Reply:
Yes, we explictly want to diffrenciate PoW from BlockHash.
They have different requirements, for example PoW doesn't even have to use a hash, just a memory-free hard to solve problem.
even SHA1 can work as PoW but not as a block hash (due to collisions)